### PR TITLE
Issue #737 Add PR template

### LINKED ISSUE #737: Add Github template for pull requests
It would be helpful to introduce a template for PRs with a checklist of things to consider/include. This would help avoiding some common mistakes developers make (including myself). For example in about half of the merged PRs, we forget to update the changelog with relevant info for the user, which then has to be added later.

Things to include in the the checklist/definition of done:

- Link to issue
- Description of changes broadly made and how they are going to fix the issue
- Update changelog, if changes are relevant to the users (e.g. bugfix, API changes, changes in behaviour)
- PR title starts with issue number, like ``#737``

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #737 

This PR adds a template for PRs. [Unfortunately the template only works if it is in the default branch](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates#pull-request-templates), (``master``) in our case, so I could not test it. I presume it works though, [as it works for Ribasim](https://github.com/Deltares/Ribasim/blob/main/.github/pull_request_template.md).

[Relevant docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+-->
+
+# Checklist
+<!---
+Before requesting review, please go through this checklist:
+-->
+
+- [ ] Links to correct issue
+- [ ] Update changelog, if changes affect users
+- [ ] PR title starts with ``#issue-nr``, e.g. #737
```
Comment (Manangka) [inline .github/pull_request_template.md:18]: We want to have in the form `Issue #nr`
e.g. `Issue #737`

**Review by Manangka — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (JoerivanEngelen) [reply .github/pull_request_template.md]: Fixed

**Review by Manangka — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Fixes #
```
Comment (luitjansl) [inline .github/pull_request_template.md:1]: what is this line for? ( Fixes #)

**Review by luitjansl — APPROVED** [review]
looks good ( 1 question)

**Pull request merged**